### ISSUE TITLE
Logic behind git advancing timestamp?

### ISSUE THREAD

**Issue by themightychris** [jgavris/rs-git-fsmonitor#10]
Does anyone understand the rhyme/reason behind the timestamp git passes the fsmonitor?

I've been monitoring its calls and trying to search/read the source and haven't been able to figure it out

It seems to start out way in the past so my whole repo is dirty which keeps git status super slow.  `git add --all` is the only thing that seems to do it consistently, but is a super slow operation. It calls the fsmonitor but doesn't seem to make use of the result as it stays super slow even after status gets fast

I'm looking for a command I can run during the workspace setup script that will do whatever needs to be done to the index so git status commands are quick for the user right after I hand control to them, but blasting it with git add --all is undesirable

`git update-index` seems to have a lot of relevant options but I haven't been able to get anything to advance the timestamp

**Comment by jgavris** [comment]
I think it's here? https://github.com/git/git/blob/master/fsmonitor.c#L130

The first status should return all the files (via the null string).

**Comment by themightychris** [comment]
Yeah I did find that but couldn't derive any practical application from it

It appears that every call to `refresh_fsmonitor` will result in the timestamp being updated, but I've observed that not every call out to the fsmonitor hook results in the timestamp being updated. So there must be thing querying the fsmonitor without "refresh"ing it.

Looking like `read-cache.c` and `dir.c` have calls to it: https://github.com/git/git/search?q=refresh_fsmonitor&unscoped_q=refresh_fsmonitor

But that still leaves the question of what git commands get down to these code paths. Ultimately I want to figure out how to force this refresh with minimal/no side-effects on repo/index state
